### PR TITLE
[SPARK-50313][SQL][TESTS] Enable ANSI in SQL *SQLQueryTestSuite by default

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR enables ANSI in SQL *SQLQueryTestSuite by default for regular tests and switches the original ansi/ ones to nonansi, following SPARK-44444, which turn on ansi for prod, 

### Why are the changes needed?

test improvements

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
new golden files

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/udf/udf-union.sql [hunk 1]
```diff
@@ -11,7 +11,7 @@ FROM   (SELECT udf(c1) as c1, udf(c2) as c2 FROM t1
 
 -- Type Coerced Union
 SELECT udf(c1) as c1, udf(c2) as c2
-FROM   (SELECT udf(c1) as c1, udf(c2) as c2 FROM t1
+FROM   (SELECT udf(c1) as c1, udf(c2) as c2 FROM t1 WHERE c2 = 'a'
```
Comment (yaooqinn) [inline sql/core/src/test/resources/sql-tests/inputs/udf/udf-union.sql:14]: stabilize error

File: sql/core/src/test/resources/sql-tests/inputs/pipe-operators.sql [hunk 2]
```diff
@@ -548,7 +548,8 @@ values (0, 'abc') tab(x, y)
 
 -- Union distinct with a VALUES list.
 values (0, 1) tab(x, y)
-|> union table t;
+|> union table t
+|> where x = 0;
```
Comment (yaooqinn) [inline sql/core/src/test/resources/sql-tests/inputs/pipe-operators.sql:552]: stabilize error

File: sql/core/src/test/resources/sql-tests/inputs/decimalArithmeticOperations.sql [hunk 3]
```diff
@@ -88,7 +88,7 @@ SELECT CAST(10 AS DECIMAL(10, 2)) div CAST(3 AS DECIMAL(5, 1));
 set spark.sql.decimalOperations.allowPrecisionLoss=false;
 
 -- test decimal operations
-select id, a+b, a-b, a*b, a/b from decimals_test order by id;
+select /*+ COALESCE(1) */ id, a+b, a-b, a*b, a/b from decimals_test order by id;
```
Comment (yaooqinn) [inline sql/core/src/test/resources/sql-tests/inputs/decimalArithmeticOperations.sql:91]: stabilize error

**Comment by yaooqinn** [comment]
cc @dongjoon-hyun as the initiator of SPARK-44444, also cc @cloud-fan, thank you

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for making this efforts, @yaooqinn . It looks reasonable to me.

Although it's too verbose due to the switching the directory locations, I believe it's okay because we need to regenerate the SQL result always when we are going to backport from `branch-4.0` to `branch-3.5`.

**Review by MaxGekk — COMMENTED** [review]
cc @mihailom-db who works on https://issues.apache.org/jira/browse/SPARK-49577 and @srielau

**Comment by yaooqinn** [comment]
Merged to master, thank you @dongjoon-hyun

**Pull request closed without merging**